tests/ensure_provider_tests.py

```diff
@@ -76,6 +76,10 @@
         MINIMUM_VERSION: "'1.25.1'",
         "comment": "# NetworkManager 1.25.1 introduced ethtool coalesce support",
     },
+    "playbooks/tests_ethtool_ring.yml": {
+        MINIMUM_VERSION: "'1.25.2'",
+        "comment": "# NetworkManager 1.25.2 introduced ethtool ring support",
+    },
```

Comment:
Get hint from
```
[wenliang@localhost NetworkManager]$ git grep -w 'NMEthtoolRingState' 1.25.1-dev 
[wenliang@localhost NetworkManager]$ git grep -w 'NMEthtoolRingState' 1.25.2-dev 
1.25.2-dev:src/devices/nm-device.c:     NMEthtoolRingState *ring;
1.25.2-dev:src/devices/nm-device.c:     gs_free NMEthtoolRingState *ring = NULL;
1.25.2-dev:src/devices/nm-device.c:     NMEthtoolRingState ring_old;
1.25.2-dev:src/devices/nm-device.c:     NMEthtoolRingState ring_new;
1.25.2-dev:src/platform/nm-platform-utils.c:                  NMEthtoolRingState *ring)
1.25.2-dev:src/platform/nm-platform-utils.c:                            NMEthtoolRingState *ring)
1.25.2-dev:src/platform/nm-platform-utils.c:                  const NMEthtoolRingState *ring)
1.25.2-dev:src/platform/nm-platform-utils.c:                            const NMEthtoolRingState *ring)
1.25.2-dev:src/platform/nm-platform-utils.h:                                     NMEthtoolRingState *ring);
1.25.2-dev:src/platform/nm-platform-utils.h:                                     const NMEthtoolRingState *ring);
1.25.2-dev:src/platform/nm-platform.c:                                   NMEthtoolRingState *ring)
1.25.2-dev:src/platform/nm-platform.c:                              const NMEthtoolRingState *ring)
1.25.2-dev:src/platform/nm-platform.h:typedef struct _NMEthtoolRingState NMEthtoolRingState;
1.25.2-dev:src/platform/nm-platform.h:                                            NMEthtoolRingState *ring);
1.25.2-dev:src/platform/nm-platform.h:                                       const NMEthtoolRingState *ring);
```

Reply:
Actually, the tests should also run for the initscripts provider since that is getting support, too. So limiting this only to NM is wrong but also wrong for the other ethtool settings. :-/

Reply:
Yes, the Ethtool Ring option was supported by initscripts[1], How do you know Coalesce option was also supported by initscripts? 

```bash
[root@localhost ~]# grep -R ETHTOOL_OPTS $(rpm -ql network-scripts)
/etc/sysconfig/network-scripts/network-functions:    for opts in $ETHTOOL_OPTS ; do
/etc/sysconfig/network-scripts/ifcfg-testnic1:ETHTOOL_OPTS="-C testnic1 adaptive-rx 1"
/etc/sysconfig/network-scripts/network-functions:    for opts in $ETHTOOL_OPTS ; do
/usr/share/doc/network-scripts/sysconfig.txt:    ETHTOOL_OPTS=...
/usr/share/doc/network-scripts/sysconfig.txt:        ETHTOOL_OPTS="speed 100 duplex full autoneg off"
/usr/share/doc/network-scripts/sysconfig.txt:      ETHTOOL_OPTS="-K ${DEVICE} tso on; -G ${DEVICE} rx 256 tx 256"
/usr/share/doc/network-scripts/sysconfig.txt:    ETHTOOL_OPTS=...
/usr/share/doc/network-scripts/sysconfig.txt:        ETHTOOL_OPTS="speed 100 duplex full autoneg off"
/usr/share/doc/network-scripts/sysconfig.txt:      ETHTOOL_OPTS="-K ${DEVICE} tso on; -G ${DEVICE} rx 256 tx 256"
```

Reply:
Take a look at the lines after

```
/etc/sysconfig/network-scripts/network-functions:    for opts in $ETHTOOL_OPTS ; do
```